resource/js/vocab-search.js

```diff
@@ -6,49 +6,246 @@ const vocabSearch = Vue.createApp({
+        const searchTermLowerCase = searchTerm.toLowerCase()
+        const labelLowerCase = label.toLowerCase()
+        if (labelLowerCase.includes(searchTermLowerCase)) {
+          const startIndex = labelLowerCase.indexOf(searchTermLowerCase)
+          const endIndex = startIndex + searchTermLowerCase.length
+          return {
+            before: label.substring(0, startIndex),
+            match: label.substring(startIndex, endIndex),
+            after: label.substring(endIndex)
+          }
+        }
+        return label
+      }
+      return null
+    },
+    translateType (type) {
+      return window.SKOSMOS.msgs[window.SKOSMOS.lang][type]
+    },
+    /*
+     * renderResults is used when the search string has been indexed in the cache
+     * it also shows the autocomplete results list
+     * TODO: Showing labels in other languages, extra concept information and such goes here
+     */
+    renderResults () {
+      // TODO: get the results list form cache if it is implemented
+      const renderedSearchTerm = this.searchTerm // save the search term in case it changes while rendering
+      this.renderedResultsList.forEach(result => {
+        if ('hiddenLabel' in result) {
+          result.hitType = 'hidden'
+          result.hit = this.renderMatchingPart(renderedSearchTerm, result.prefLabel)
+        } else if ('altLabel' in result) {
+          result.hitType = 'alt'
+          result.hit = this.renderMatchingPart(renderedSearchTerm, result.altLabel)
+          result.hitPref = this.renderMatchingPart(renderedSearchTerm, result.prefLabel)
+        } else if ('prefLabel' in result) {
+          result.hitType = 'pref'
+          result.hit = this.renderMatchingPart(renderedSearchTerm, result.prefLabel)
+        }
+        if ('uri' in result) { // create relative Skosmos page URL from the search result URI
+          result.pageUrl = window.SKOSMOS.vocab + '/' + window.SKOSMOS.lang + '/page?'
+          const urlParams = new URLSearchParams({ uri: result.uri })
+          result.pageUrl += urlParams.toString()
+        }
+        // render search result renderedTypes
+        if (result.type.length > 1) { // remove the type for SKOS concepts if the result has more than one type
+          result.type.splice(result.type.indexOf('skos:Concept'), 1)
+        }
+        // use the translateType function to map translations for the type IRIs
+        result.renderedType = result.type.map(this.translateType).join(', ')
+      })
+
+      if (this.renderedResultsList.length === 0) { // show no results message
+        this.renderedResultsList.push({
+          prefLabel: this.msgs['No results'],
+          lang: window.SKOSMOS.lang
+        })
+      }
+      const element = document.getElementById('search-autocomplete-results')
+      element.classList.add('show')
+    },
+    hideDropdown () {
+      const element = document.getElementById('search-autocomplete-results')
+      element.classList.remove('show')
```

Comment:
Instead of removing class directly, I would use a variable to control dropdown visibility, for example:
```suggestion
      this.showDropdown = false
```